### ISSUE TITLE
[DOCUMENTATION] Add new admonitions to telegram classes

### ISSUE THREAD

**Issue by harshil21** [python-telegram-bot/python-telegram-bot#3496]
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

I think the docs can do with more explicit types of admonitions. I have a few suggestions:

1. "Returned in":  When viewing a documentation of a class, sometimes I forget which bot method or object through which this class is available.

<details>
<summary> Examples </summary>
<br>

``` python
class ChatInviteLink:
    """
    .. admonition:: Returned In
        :class: returned-in

        :meth:`Bot.create_chat_invite_link`, ...
    """
```

</details>

2. "Available in": Similar to the above, but this would include where and how this object can be found. Motivation: Our docs are currently top-down, i.e. Update -> Message -> Chat, so if I landed in "Update" I can figure out how to get the Chat object, but not vice-versa.  Icon suggestions: [Box-fill](https://icons.getbootstrap.com/icons/box-fill/), [Boxes](https://icons.getbootstrap.com/icons/boxes/), [Geo fill](https://icons.getbootstrap.com/icons/geo-fill/) (liked more by me)

<details>
<summary> Examples </summary>
<br>

``` python
class Chat:
    """
    .. admonition:: Available in
        :class: available-in

        :attr:`telegram.Message.chat`, ...
    """

class Message:
    """
    .. admonition:: Available in
        :class: available-in

        :attr:`telegram.Update.message`, ...
    """
```

</details>

3. "Use in": This would be used in classes which need to be instantiated and then passed into bot methods parameters.
Icon suggestions: [Arrow left](https://icons.getbootstrap.com/icons/arrow-left/), [Paperclip](https://icons.getbootstrap.com/icons/paperclip/), [Funnel](https://icons.getbootstrap.com/icons/funnel/) (liked more by me), [Stack](https://icons.getbootstrap.com/icons/stack/)

<details>
<summary> Examples </summary>
<br>

``` python
class ChatPermissions:
    """
    .. admonition:: Used in
        :class: used-in

        :meth:`Bot.set_chat_permissions`, ...
    """
```

</details>

4. "Shortcuts": This would be used in the bot methods instead of the See Also currently used.

### Describe the solution you'd like

These admonitions are better than "see also" (in most cases) since it tells the user what and why it was linked in the first place.

### Describe alternatives you've considered

see alsos

### Additional context

How to make custom admonitions: https://pradyunsg.me/furo/reference/admonitions/

**Comment by Bibo-Joshi** [comment]
Just to clarify: `Returned In` and `Available in` are new cross-references and `Shortcuts` would be extracting the links from the `.. seealso` directives, right?

Another thought: Maybe we can automate the generation of these admonitions? 🤔

**Comment by harshil21** [comment]
> Just to clarify: `Returned In` and `Available in` are new cross-references and `Shortcuts` would be extracting the links from the `.. seealso` directives, right?

that's correct.

> Another thought: Maybe we can automate the generation of these admonitions? 🤔

would be great. Or maybe even a way to verify them

**Comment by lemontree210** [comment]
If you want to automate then I'm not sure I'm up to the task (especially given that you surely want to check return types, not just check for "Returned in" strings in docstrings), but if it's my good old documentation task, then I can start digging in :)

**Comment by harshil21** [comment]
I think we should first try to automate it, mostly because the number of cross references is increasing like crazy, and we can also not verify every single one of them all the time. But I think the Shortcut admonition could be done now since I'm not sure how that would be automated (shortcuts usually have different names altogether)

**Comment by lemontree210** [comment]
OK. Well theoretically one could look for "Shortcut for" in docstrings of shortcut classes, then go to the definition in `Bot` and add a custom admonition.

**Comment by harshil21** [comment]
yes, also just realized we can check the method source code for calls to a bot method... anyway doing the Shortcut thing is not super high effort so could be done now

**Comment by lemontree210** [comment]
I'll be doing the Shortcuts then

**Comment by Bibo-Joshi** [comment]
> yes, also just realized we can check the method source code for calls to a bot method... 

That's what I had in mind. 

For return type, we can either check the docstring or the return annotation - I prefer the latter one.

**Comment by lemontree210** [comment]
>     1. "Returned in":  When viewing a documentation of a class, sometimes I forget which bot method or object through which this class is available.

Do we want to list that for "popular" classes as well? E.g. all methods where `Message` is returned

**Comment by Bibo-Joshi** [comment]
@lemontree210 that&#x27;s the idea, yes

**Comment by lemontree210** [comment]
is it OK to put the "returned in" admonition here?

```
    """This object represents a chat.

    .. admonition:: Returned in
        :class: returned-in
        
        :meth:`telegram.Bot.get_chat`

    Objects of this class are comparable in terms of equality. Two objects of this class are
    considered equal, if their :attr:`id` is equal.
```

**Comment by lemontree210** [comment]
where do `..versionadded` or `..versionchanged` notes go? before or after cross-references?

**Comment by lemontree210** [comment]
For "Available in", do I analyze typing of attributes of each class?

**Comment by Bibo-Joshi** [comment]
@lemontree210 That&#x27;s probably an option, though I&#x27;m now sure how well that works. Haven&#x27;t fiddled with <code>typing.get_type_hints</code> yet. The alternative would be to parse the docstrings …

**Comment by harshil21** [comment]
Realized that this just leaves out one more admonition: "Use in". We can add these in classes like `BotCommandScope` and `ChatPermissions`. Can also be automated. Will update OP

**Comment by Bibo-Joshi** [comment]
closed by #3501

**Comment by lemontree210** [comment]
Sorry, I'll allow myself to re-open it :) I think that @harshil21 also suggested automating the "Shortcuts" as well. Right now they're manual.

**Comment by lemontree210** [comment]
closed with #3529
